x-pack/platform/plugins/private/intercepts/public/prompter/component/intercept_display_manager/intercept_display_manager.tsx

```diff
@@ -0,0 +1,289 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import * as Rx from 'rxjs';
+import React, { useEffect, useState, useCallback, useMemo, useRef } from 'react';
+import { i18n } from '@kbn/i18n';
+import { css } from '@emotion/react';
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiPortal,
+  useEuiTheme,
+  EuiTitle,
+  EuiTourStepIndicator,
+  euiFlyoutSlideInRight,
+  euiCanAnimate,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiSplitPanel,
+  EuiForm,
+  EuiButtonIcon,
+} from '@elastic/eui';
+import type { HttpStart } from '@kbn/core-http-browser';
+import type { EuiTourStepStatus } from '@elastic/eui/src/components/tour/tour_step_indicator';
+import { InterceptDialogApi } from '../../service/intercept_dialog_api';
+
+export type Intercept = Rx.ObservedValueOf<ReturnType<InterceptDialogApi['get$']>>[number];
+
+interface InterceptDialogManagerProps {
+  ackIntercept: (args: Parameters<InterceptDialogApi['ack']>[0] & Pick<Intercept, 'runId'>) => void;
+  /**
+   * Observable that emits the intercept to be displayed
+   */
+  intercept$: Rx.Observable<Intercept>;
+  /**
+   * Helper to load static assets pertinent to the intercept plugin
+   */
+  staticAssetsHelper: HttpStart['staticAssets'];
+}
+
+interface InterceptProgressIndicatorProps {
+  stepsTotal: number;
+  currentStep: number;
+}
+
+const InterceptProgressIndicator = React.memo(
+  ({ stepsTotal, currentStep }: InterceptProgressIndicatorProps) => {
+    if (!stepsTotal) return null;
+
+    return (
+      <EuiFlexItem grow={false}>
+        <ul className="euiTourFooter__stepList">
+          {[...Array(stepsTotal).keys()].map((_, i) => {
+            let status: EuiTourStepStatus = 'complete';
+            if (currentStep === i) {
+              status = 'active';
+            } else if (currentStep <= i) {
+              status = 'incomplete';
+            }
+            return <EuiTourStepIndicator key={i} number={i + 1} status={status} />;
+          })}
+        </ul>
+      </EuiFlexItem>
+    );
+  }
+);
+
+export function InterceptDisplayManager({
+  ackIntercept,
+  intercept$,
+  staticAssetsHelper,
+}: InterceptDialogManagerProps) {
+  const interceptRenderMark = useRef<PerformanceMark>();
+  const { euiTheme } = useEuiTheme();
+  const [currentStepIndex, setCurrentStepIndex] = useState(0);
+  const [currentIntercept, setCurrentIntercept] = useState<Intercept | null>(null);
+  const feedbackStore = useRef<Record<string, unknown>>({});
+  const startIllustrationWidth = useRef(89); // Magic number was provided by Ryan
```

Comment:
nit: avoid hard-coded magic numbers in code where they can get lost. Prefer declaring in a constants file.